analyses/upstream-analysis/util/function-process-Seurat.R

```diff
@@ -0,0 +1,322 @@
+#############################################################################################
+#' Function to calculate cell cycle scores (S and G2M) for each cell.
+#' Genome specifies which gene set to use for cell cycle genes
+#' @param seurat_obj
+#' @param Genome
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' 
+Genome_Specific_Cell_Cycle_Scoring <- function(seurat_obj, Genome) {
```

Comment:
For this function, if the GRCh38ANDGRCm39 dual genome is used for the analysis, this will be another place where uses of mm10 should likely be updated to GRCm39 since the gene names in that dual reference are annotated as GRCm39_<gene name>. Additionally, it seems those gene names are named with _ not - (e.g. GRCh38_<gene name> or GRCm39_<gene name>) in the reference files (e.g. fasta, gtf).

A broader/general note for future consideration, maybe this function is one where reducing hard coding of the genome prefixes might be useful in the long run, if people end up running the module with different genomes than those that the core provides or if the genomes are updated again and the prefix annotations change (although that may not happen for quite a while as GRCm39 is a new release)

Reply:
I am unsure if the `Dualhg19`, `DualGRCh38`, and `DualGRCm39` also need to be changed and specified further, or how to remove the hardcoded part. Any thoughts @CodyRamirez?

Reply:
This portion of the code is to address cellranger counts gene naming for genomes to ensure that the correct style of naming is being used when identifying cell cycle genes. When cellranger aligns to a human genome it follows the typical nomenclature: Human genes have all letters in upper case. When cellranger aligns to a mouse genome it follows the typical nomenclature: Mice have the first letter capitalized followed by lower case letter.

However, when cellranger aligns to a dual genome it follows the same above naming nomenclature but it appends the specific genome to the genes. For example, any human genes will now be reference_genome--gene (i.e. hg19--TP53, hg19--RHO, mm10--Trp53, mm10--Rho). So the function needs you to tell it which reference genome you used to ensure it uses the correct cell cycle gene to score the data.